tests/dbshell/test_sqlite.py

```diff
@@ -21,3 +25,19 @@ def test_parameters(self):
             self.settings_to_cmd_args_env({"NAME": "test.db.sqlite3"}, ["-help"]),
             (["sqlite3", "test.db.sqlite3", "-help"], None),
         )
+
+    def test_non_zero_exit_status_when_path_to_db_is_path(self):
+        sqlite_with_path = {
+            "ENGINE": "django.db.backends.sqlite3",
+            "NAME": Path("test.db.sqlite3"),
+        }
+        cmd = DatabaseClient.settings_to_cmd_args_env(sqlite_with_path, [])[0]
+
+        def raise_error(*args, **kwargs):
+            raise subprocess.CalledProcessError(returncode=1, cmd=cmd)
+
+        expected_message = '"sqlite3 test.db.sqlite3" returned non-zero exit status 1.'
+        with mock.patch(
+            "django.db.backends.sqlite3.client.DatabaseClient.runshell", new=raise_error
```

Comment:
We can raise directly with patch like so (tested locally):

```suggestion
        expected_message = '"sqlite3 test.db.sqlite3" returned non-zero exit status 1.'
        with mock.patch(
            "django.db.backends.sqlite3.client.DatabaseClient.runshell",
            side_effect=subprocess.CalledProcessError(returncode=1, cmd=cmd),
```